### PR TITLE
When rejected pegout is refunded, post RSKIP427 refund the total value sent in weis

### PR DESCRIPTION
**Pull request by marcos-iov**
- Change pegout refund logic in BridgeSupport to return the total value sent to the Bridge post RSKIP427
- Add tests

### CODE REVIEW

**Review by nathanieliov — APPROVED** [review]
Well done. Left some suggestions

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 1]
```diff
@@ -1274,15 +1276,22 @@ void low_amount_release_request_rejected_before_lovell_value_in_satoshis() throw
 
         Coin belowPegoutMinimumValue = BRIDGE_CONSTANTS.getMinimumPegoutTxValue().minus(Coin.SATOSHI);
         co.rsk.core.Coin pegoutRequestValue = co.rsk.core.Coin.fromBitcoin(belowPegoutMinimumValue);
+        // Add some extra weis to the value, but less than 1 satoshi.
+        // To ensure that the pegout value is rounded down to fit in satoshis.
+        co.rsk.core.Coin oneSatoshiInWeis = co.rsk.core.Coin.valueOf(Denomination.satoshisToWeis(1).longValue());
```
[the remark this answers is not in the document]
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:1281]: Good idea, thanks!

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 2]
```diff
@@ -1274,15 +1276,22 @@ void low_amount_release_request_rejected_before_lovell_value_in_satoshis() throw
...
+        co.rsk.core.Coin extraWeis = oneSatoshiInWeis.subtract(co.rsk.core.Coin.valueOf(1));
```
[the remark this answers is not in the document]
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:1282]: Not sure about this one. `Denomination.WEI` isn't really clear to me if that represents 1 wei or what. `valueOf(1)` is quiete explicit

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 3]
```diff
@@ -1274,15 +1276,22 @@ void low_amount_release_request_rejected_before_lovell_value_in_satoshis() throw
 
         Coin belowPegoutMinimumValue = BRIDGE_CONSTANTS.getMinimumPegoutTxValue().minus(Coin.SATOSHI);
         co.rsk.core.Coin pegoutRequestValue = co.rsk.core.Coin.fromBitcoin(belowPegoutMinimumValue);
+        // Add some extra weis to the value, but less than 1 satoshi.
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:1279]:
```suggestion
        // Add one extra wei to the value, that is less than 1 satoshi.
```
wdyt?

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 4]
```diff
@@ -1274,15 +1276,22 @@ void low_amount_release_request_rejected_before_lovell_value_in_satoshis() throw
...
+        // To ensure that the pegout value is rounded down to fit in satoshis.
+        co.rsk.core.Coin oneSatoshiInWeis = co.rsk.core.Coin.fromBitcoin(Coin.SATOSHI);
+        co.rsk.core.Coin extraWeis = oneSatoshiInWeis.subtract(co.rsk.core.Coin.valueOf(1));
+        pegoutRequestValue = pegoutRequestValue.add(extraWeis);
 
         bridgeSupport.releaseBtc(buildReleaseRskTx(pegoutRequestValue));
 
         RskAddress senderAddress = new RskAddress(SENDER.getAddress());
 
+        // Pre lovell, the refund should be rounded down to the nearest satoshi.
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:1289]: do you think this comment is necessary even having the other one [here](https://github.com/rsksmart/rskj/pull/2798/files#diff-e76588191a0bacc0270da08e9e8242cb6eb1c26b051a9e76ad7bb4e60376fafcR1264)?

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 5]
```diff
@@ -1299,12 +1308,15 @@ void low_amount_release_request_rejected_before_lovell_value_in_satoshis() throw
         CallTransaction.Function event = BridgeEvents.RELEASE_REQUEST_REJECTED.getEvent();
         assertArrayEquals(event.encodeSignatureLong(), firstLog.getTopics().get(0).getData());
 
+        // Same case in the log, the value should be rounded down to the nearest satoshi.
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:1311]: same here

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 6]
```diff
@@ -1274,15 +1276,22 @@ void low_amount_release_request_rejected_before_lovell_value_in_satoshis() throw
 
         Coin belowPegoutMinimumValue = BRIDGE_CONSTANTS.getMinimumPegoutTxValue().minus(Coin.SATOSHI);
         co.rsk.core.Coin pegoutRequestValue = co.rsk.core.Coin.fromBitcoin(belowPegoutMinimumValue);
+        // Add some extra weis to the value, but less than 1 satoshi.
+        // To ensure that the pegout value is rounded down to fit in satoshis.
+        co.rsk.core.Coin oneSatoshiInWeis = co.rsk.core.Coin.fromBitcoin(Coin.SATOSHI);
+        co.rsk.core.Coin extraWeis = oneSatoshiInWeis.subtract(co.rsk.core.Coin.valueOf(1));
```
[the remark this answers is not in the document]
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java]: That sounds good, thanks!

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java (hunk 4 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:1290]: I actually do yes. I think we tend to focus on what's right in front of us

**Pull request merged**